src/backend/services/wallet-api/src/routes.py

```diff
@@ -31,6 +36,39 @@ def getWalletAmount():
     response = send_get_request( blockchain_wallet_url.format("balance"), data)
     return jsonify(response.json()), response.status_code
 
+@cross_origin()
+@app.route("/history", methods=['GET'])
+def getWalletHistory():
+
+    umnetId = request.args.get('umnetId')
+
+    if umnetId is None:
+        raise IncorrectPayloadException()
+
+    response = requests.get( blockchain_url.format("chain")).json()
+    chain = response['chain']
+
+    result = []
+
+    for block in chain[1:]:
+        block = json.loads(block)
+        ts = float(block["timestamp"])
```

Comment:
block timestamp was removed, access the transaction's timestamp.

Reply:
done